_includes/manuals/nightly/3.7_migration.md

```diff
@@ -0,0 +1,171 @@
+
+    foreman-rake reports:expire days=2 batch_size=50000 sleep_time=0.000001 RAILS_ENV="production"
+
+The command above deletes records in batches of 50k items, it can take a lot of
+time to process all records. In both cases, progress can be seen in the
+`production.log`.
+
+#### Migrating
+
+To migrate the data from one database to the other, we will use a rake task.
+We will be using `foreman-rake db:convert:prod2dev` to migrate the data.
+
+Foreman internally uses a development framework called Rails. In Rails, you can
+have a specific configuration for e.g. a development and a production
+environment. Our prod2dev rake task assumes, that the source database is
+configured as the production database and the new target database is configured
+as the development database. All data in the development database will be
+overwritten.
+
+You can configure the new database as Foreman's new development database by
+adding a new connection in `/etc/foreman/database.yml`.
+
+It may look like this:
+
+```yaml
+# The new target database
+development:
+  adapter: postgresql
+  host: postgres.example.com
+  port: 5432
+  database: foreman
+  username: foreman
+  password: "foreman"
+  pool: 25
+
+# The old source database
+production:
+  adapter: mysql2
+  host: mysql.example.com
+  port: 3306
+  database: foreman
+  username: foreman
+  password: "foreman"
+  pool: 25
+```
+
+In the next step, call a rake task which will create new database:
+
+```sh
+forman-rake db:create RAILS_ENV=development
+```
+
+The new database needs to be set up with all the tables required for Foreman to
+operate, so in order to prepare the new database to receive the actual data, we
+need to run all of Foreman's database migrations.
+
+```sh
+foreman-rake db:migrate RAILS_ENV=development
+```
+
+Before proceeding with the actual data migration, it's required to do separate
+migration of dynflow tables.
+
+```sh
+foreman-rake dynflow:migrate
+```
+
+Next, we can finally migrate all the data. This will take a long time depending
+on the size of your database. It is highly recommended to run this in a screen
+or tmux session.
+
+```sh
+forman-rake db:convert:prod2dev
+
+```
+
+Postgresql uses sequences in order to store the next free primary key id for a
+table. These sequences do not get updated by the rake tasks and have to be
+adjusted manually.
+
+```sh
+forman-rake sequence:reset RAILS_ENV=development
+
+```
+
+#### Wrapping up
+
+Then, make sure to resore the original permissions of your foreman database
+user. This can be done on Postgresql by issuing this SQL command.
+
+```sql
+ALTER ROLE "foreman" NOSUPERUSER;
+```
+
+To switch your Foreman instance to the new database, you need to change the
+production database configuration in `/etc/foreman/database.yml` to point to
+the development installation we configured earlier. You can either do this
+manually or re-run foreman-installer with the new parameters for Postgresql.
```

Comment:
It's not clear what needs to be changed here if done manually or what are the new installer parameters for postgresql (and what parameters need to be reset to remove the mysql settings)

Reply:
Added bunch of instructions.